source/common/router/router.cc

```diff
@@ -869,7 +892,7 @@ void Filter::onSoftPerTryTimeout(UpstreamRequest& upstream_request) {
     RetryStatus retry_status =
         retry_state_->shouldHedgeRetryPerTryTimeout([this]() -> void { doRetry(); });
 
-    if (retry_status == RetryStatus::Yes && setupRetry()) {
+    if (retry_status == RetryStatus::Yes) {
       setupRetry();
```

Comment:
curious, did we double count stats here?

Reply:
Yes, I think so; I found this while investigating this a few months ago. See the Fixed issue in description.